### DISCUSSION TITLE
How do i test these APIs for ONDC, Where are they hosted?

### DISCUSSION THREAD

**Discussion by deepanshurathi553** [beckn/protocol-specifications#258] [Q&A]
I tried searching everywhere but could not find any, How can I as a software engineer test these APIs?
Do i have to write my own server where I subscribe to endpoints like "on_search" If yes, Where do i register my APIs?

**Comment by venkatramanm** [comment]
If you want to play around you can use the registry of beckn open collective, "https://registry.becknprotocol.io" you can self register your endpoints. 
if you want to test in ONDC, please refer to ondc github for the process.. You need to send  a mail to some one in Protean(former nsdl) to add you to the staging registry. But these are Serious places for participants who are trying to get online to ondc. If you have an application and wish to onboard onto ondc, by all means get on the staging registry or if you want to play around and make  your own apps to try out etc, feel free to use the beckn open collective's registry. 

Also I suggest you join the discord channel of beckn https://discord.gg/ejeMtuG2Gs  also. Some of us are quite active in responding there. :)

**Reply by Amealsuraj** [reply]
I want to test all APIs without registration in Staging Registry. How can I test?

**Comment by rahulbansal16** [comment]
@venkatramanm is it possible for the registry to selectively deny access to some of the BPP?

**Reply by venkatramanm** [reply]
Question is not very clear. But let me try to clarify some doubts. Registry is just a place where participant's roles , url, id, key etc are stored for others to lookup. IF for any reason the url is offline or the keys are not working during /on_subscribe, the  entry would not go to "SUBSCRIBED" state. 
During /search, it is gateway that decides bpps to broadcast. There is no incentive for gw to block some bpp unless it is not reachable and waiting for ACK etc just times out and people spend time waiting.! In such cases, a gw may temporarily disable but that is a choice of the gw.
